### PR TITLE
[Badge]

### PR DESCRIPTION
**Pull request by ignasvalotka**
image

### CODE REVIEW

File: ios/Sources/MaaS/UI/Components/Badge.swift [hunk 1]
```diff
@@ -1,65 +1,309 @@
...
+
     public struct InputType {
-        let color: Color?
+        let type: BadgeType
+        let backgroundColor: Color?
+        let foregroundColor: Color?
         let icon: UIImage?
         let text: String?
+        let alternativeBadgesColors: [Color]
+        let subBadge: UIImage?
```
Comment (justasm) [inline ios/Sources/MaaS/UI/Components/Badge.swift:22]: @stonevyt called this `subbadgeIcon` in Compose, would be great to strive for consistency 🌟

**Review by justasm — COMMENTED** [review]
Looks good overall to me, would be nice to hear some input from iOS colleagues regarding SwiftUI 🙂

File: ios/Sources/MaaS/UI/Components/Badge.swift (hunk 1 above)
Comment (ignasvalotka) [reply ios/Sources/MaaS/UI/Components/Badge.swift:22]: Yes! missed this one

File: ios/Sources/MaaS/UI/Components/Badge.swift [hunk 2]
```diff
@@ -1,65 +1,309 @@
...
+            }
+        }
+        .fixedSize()
+        .subBadge(image: input.base.subBadgeIcon,
+                  width: constants.subBadgeIconWidth,
+                  height: constants.subBadgeIconHeight)
+    }
+
+    private var badges: [Badge.InputType] {
+        let maxAlternativeBadges = 2
```
Comment (mikasdo) [inline ios/Sources/MaaS/UI/Components/Badge.swift:223]: This should be a shared constant

File: ios/Sources/MaaS/UI/Components/Badge.swift [hunk 3]
```diff
@@ -1,65 +1,309 @@
...
+        let alternativeBadges = Array(input.colors.prefix(maxAlternativeBadges)).toAlternativeBadges()
+        return alternativeBadges + [input.base]
+    }
+
+    private func spacing(index: Int) -> CGFloat {
+        CGFloat(2.5 * Double(index))
```
Comment (mikasdo) [inline ios/Sources/MaaS/UI/Components/Badge.swift:229]: This should probably also be a constant

Comment (ignasvalotka) [reply ios/Sources/MaaS/UI/Components/Badge.swift:229]: Yes! This actually can be computed from the constants we already have!

File: ios/Sources/MaaS/UI/Components/Badge.swift (hunk 2 above)
Comment (ignasvalotka) [reply ios/Sources/MaaS/UI/Components/Badge.swift:223]: Agreed

File: common/core/src/commonMain/kotlin/com/trafi/ui/theme/internal/Color.kt [hunk 4]
```diff
@@ -28,12 +28,3 @@ internal fun Color.alpha(value : Float): Color {
 }
 
 internal fun Color(color: Long) = Color((color.toULong() and 0xffffffffUL) shl 32)
-
-fun String?.toColor(fallback: Long = 0xFF_000000): Color {
```
Comment (justasm) [inline common/core/src/commonMain/kotlin/com/trafi/ui/theme/internal/Color.kt:32]: This shouldn't be removed

**Review by mikasdo — APPROVED** [review]

**Pull request merged**